File: javascript/src/models/sinkOtelV1Config.ts
// this file is @generated

export interface SinkOtelV1Config {
  url: string;
  headers?: { [key: string]: string };
}

export const SinkOtelV1ConfigSerializer = {
  _fromJsonObject(object: any): SinkOtelV1Config {
    return {
      url: object["url"],
      headers: object["headers"],
    };
  },

  _toJsonObject(self: SinkOtelV1Config): any {
    return {
      url: self.url,
      headers: self.headers,
    };
  },
};

File: rust/src/models/sink_otel_v1_config.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, PartialEq, Deserialize, Serialize)]
pub struct SinkOtelV1Config {
    pub url: String,

    #[serde(skip_serializing_if = "Option::is_none")]
    pub headers: Option<std::collections::BTreeMap<String, String>>,
}

impl SinkOtelV1Config {
    pub fn new(url: String) -> Self {
        Self { url, headers: None }
    }
}
